co-log-core/src/Colog/Core/Class.hs

```diff
@@ -18,11 +17,12 @@ TODO: laws
 class HasLog env msg m where
     getLogAction :: env -> LogAction m msg
     setLogAction :: LogAction m msg -> env -> env
+    setLogAction = seq
```

Comment:
Why `seq` instead of `flip const`? Why are we forcing evaluation of the LogAction?

Reply:
sorry, I'm not sure I'm following, aren't `seq` and `flip const` equal?
```
>> :t seq
seq :: a -> b -> b
>> :t flip const
flip const :: b -> c -> c
```

Reply:
`setLogAction` here should be implemented using `overLogAction`.

Also, don't forget to add `{-# INLINE #-}` pragmas to the default implementations 🙂